docker/jupyter/custom/jupyter_delocalize.py

```diff
@@ -158,3 +159,61 @@ def delete_file(self, path):
     self._log_and_call_file_cmd_async([
         'rm', self._remote_path(meta, os_path),
     ])
+
+
+class WelderContentsManager(FileContentsManager):
+  """
+  A contents manager which integrates with the Leo Welder service.
+
+  Blocking Welder API calls are made before files are persisted. After a
+  successful call to Welder, files are persisted to the local Jupyter file
+  system as usual.
+  """
+
+  def __init__(self, *args, **kwargs):
+    # This log line shouldn't be necessary, but Jupyter's built-in logging is
+    # lacking and its configuration can be complex. Having this in the server
+    # logs is useful for confirming which ContentsManager is in use.
+    self.log.info('initializing WelderContentsManager')
+    super(WelderContentsManager, self).__init__(*args, **kwargs)
+
+  def _welder_delocalize(self, path):
+    # Ignore storage link failure, throw other errors.
+    resp = requests.post('http://localhost:8080/objects', data=json.dumps({
+      'action': 'safeDelocalize',
+      # Sometimes the Jupyter UI provided "path" contains a leading /, sometimes
+      # not; strip for Welder.
+      'localPath': path.lstrip('/')
+    }))
+    if resp.status_code != requests.codes.ok:
```

Comment:
does `requests.code.ok` include 204? I recently changed return code to 204...The fact it was 200 previously was an oversight

Reply:
Good catch, done.